VAULT-31525 Remove lowercasing of RADIUS auth.

JMGoldsmith opened the pull request:
### Description
What does this PR do?

This PR will change the behavior of the RADIUS authentication plugin for Vault. It will now allow for users with capital letters in their usernames to be used within Vault. Previous behavior would return an all lower case user, which would be rejected by the RADIUS server if it was expecting mixed case or all caps.

This was changed due to user reports of issues with capitalization.

Existing usernames that were all lower case will not be affected, but if you use mixed case or all capitals, you will now be able to properly authenticate.

### TODO only if you're a HashiCorp employee
- [x] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [x] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

fairclothjm commented on changelog/28884.txt:
````diff
@@ -0,0 +1,3 @@
+```release-note:bug
+auth/radius: Fixed and issue where upper case items were returning as lower case.
````
```suggestion
auth/radius: Fixed an issue where usernames with upper case characters where not honored
```

We may also want to add a `release-note:change` since this is a behavior change that will also result in changes to how policies are assigned

fairclothjm commented:
I am not so sure about backporting this. This is a behavior change that could result in negative effects for existing users. RE: usernames are now case sensitive and vault policies may be assigned differently.

JMGoldsmith commented:
> I added the missing `backport/1.18.x` label in case we do decide to backport this. But I don't think we should.

I've removed all backports as I agree. I will be adding a "Known Issue" PR separately per internal documentation that will be exposed to customers in 1.19 once the release notes are added to the docs. That PR should be up today/tomorrow.

fairclothjm commented:
I added the 1.19-rc milestone since we decided not to backport this

fairclothjm commented:
Thanks @JMGoldsmith ! Great work!
